### PR TITLE
#345: Allowed all branch names

### PR DESCRIPTION
**Pull request by siredmar**
- allowed branch names
- modified changelog for 0.3.0

### CODE REVIEW

**Review by mlilien — COMMENTED** [review]
@siredmar With this change, two different branches can create the same version, am i right? If so, we shouldnt merge it.

**Comment by siredmar** [comment]
@mlilien This was also the behaviour before this merge. Think about the following scenario
repo Tag 0.1.0
-> branch `feature/test1`
-> branch `feature/test2`

If gitversion is run in either of those branches it will create 0.1.1. gitversion cannot know that there is a parallel branch that is also based on 0.1.0. 
However, gitversion also creates several more verbose version information containing the branch name. Maybe we should use this one instead e.g. `GitVersion_InformationalVersion=0.2.1-test-integration.1+2.Branch.test-integration.Sha.4858dab8ce31016e87c08b3131405fe15072b961`?

@mlilien @FlorianReinhold @JoergZeidler 
Currently we don't have any process that defines the workflow of generating new releases. 
As gitversion always takes the last set SemVer tag in the repo to create a new version, we should define a process of creating those tags.
My proposal:
1. Do your changes in a PR containing Changelog modifications.
2. Merge the PR after review to `master`
3. Tag the new master branch e.g. 0.2.1 (if previous was 0.2.0)
3. (**Either locally or**) by CI/CD build the release `triggered by tag`

With this proposal it is possible to generate several versions e.g. 0.2.2. We previously discussed that making a release by tagging is a manual step. Intermediate commit between releases won't create a release, because no new tag was set (unless we define some test-tags or developemt-tags).
So what do you think?

**Review by mlilien — APPROVED** [review]

**Pull request merged**
